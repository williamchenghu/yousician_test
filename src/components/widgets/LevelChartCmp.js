import React from 'react'

import PieChart from 'react-minimal-pie-chart'
import PropTypes from 'prop-types'

export const LevelCharCmp = ({ level }) => {
  return (
    <div className="donut-level">
      <PieChart
        className="donut"
        data={[
          { value: level, key: 1, color: '#4fc514' },
          { value: 15 - level, key: 3, color: '#878789' },
        ]}
        lineWidth={10}
        paddingAngle={5}
        lengthAngle={-360}
        animate={true}
      />
      <div className="number">{level}</div>
    </div>
  )
}
LevelCharCmp.propTypes = {
  level: PropTypes.number.isRequired,
}
