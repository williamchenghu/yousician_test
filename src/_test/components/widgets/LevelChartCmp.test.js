import React from 'react'
import LevelChartCmp from '../../../components/widgets/LevelChartCmp'

describe('Cmp: LevelChartCmp', () => {
  const props = {
    level: 5,
  }
  describe('activate: LevelChartCmp', () => {
    it('should be defined', () => {
      // the component should be define
      expect(LevelChartCmp).toBeDefined()
    })

    it('should render correctly', () => {
      //prepare
      let wrapper = shallow(<LevelChartCmp {...props} />)
      //check
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
})
