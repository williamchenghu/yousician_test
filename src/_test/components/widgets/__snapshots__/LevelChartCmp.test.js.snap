// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cmp: LevelChartCmp activate: LevelChartCmp should render correctly 1`] = `
<div
  className="donut-level"
>
  <ReactMinimalPieChart
    animate={true}
    animationDuration={500}
    animationEasing="ease-out"
    className="donut"
    cx={50}
    cy={50}
    data={
      Array [
        Object {
          "color": "#4fc514",
          "key": 1,
          "value": 5,
        },
        Object {
          "color": "#878789",
          "key": 3,
          "value": 10,
        },
      ]
    }
    lengthAngle={-360}
    lineWidth={10}
    paddingAngle={5}
    radius={50}
    ratio={1}
    rounded={false}
    startAngle={0}
  />
  <div
    className="number"
  >
    5
  </div>
</div>
`;
